riff-raff/public/docs/riffraff/administration/auth.md

```diff
@@ -8,6 +8,19 @@ Authentication
 Authentication is handled by Google. This obtains the users full name and e-mail address and both of these are used to 
 provide an audit trail of deployments and other actions in the app.
 
+Google auth is implemented in two layers
+- Load Balancer layer: this component is defined in [the main riff raff cloudformation template](https://github.com/guardian/deploy-tools-platform/blob/main/cloudformation/riffraff/riffraff.template.yaml) and uses [authenticate-oidc](https://docs.aws.amazon.com/elasticloadbalancing/latest/application/listener-authenticate-users.html#configure-user-authentication). The template relies on a secret stored in SSM.
+- Application layer: this layer uses The Guardian's [Play Google Auth Module](https://github.com/guardian/play-googleauth) and is configured in the main [application.conf](https://github.com/guardian/riff-raff/blob/afb7e602e11acd7a07aae433c74be22976d8a7cd/riff-raff/conf/application.conf#L40-L41).
+
+To rotate the secrets, follow these steps:
+1. Create a new Client Secret in Riff Raff's Google Cloud project
+2. Patch the PROD and CODE application.conf to use the new secret
+3. Update the secret in AWS Secrets Manager used by the cloudformation template.
+4. Use riff raff to deploy itself, which will re-run the application startup routine (including loading the updated `application.conf`) while keeping the same instance alive
+5. Disable the old credentials in the Google Cloud Project
+6. Wait for at least 2 or 3 hours to confirm the old credentials are no longer being used, as the turn around time for disabling credentials is slow and not deterministic
```

Comment:
Instead of waiting, we could log into Riff-Raff via an incognito window to prove that logging in (via both layers of auth) still works with the new secret?

Reply:
I've updated this in https://github.com/guardian/riff-raff/pull/1445/commits/1a2ff0cf71154b1e6ac69bc96bb549b1e3d91e29.